### PR TITLE
Cleanup & update ErrorRecord class code style

### PR DESCRIPTION
**Pull request by vexx32**
# PR Summary

Cleanup and update ErrorRecord class code style & improve consistency of style across the class.

There are no _functional_ changes to this PR; everything is purely code-style-related. Some private properties were ditched entirely, being easily replaced with simple auto-properties (i.e., they just had a straightforward get/set anyway, and the extra code was adding complexity and nothing else).

## PR Context

After speaking with @SteveL-MSFT on Twitter, I intend to implement a Suggestion property to ErrorRecord to replace the current Suggestions system.

Before that, I would like to clean up the ErrorRecord class first.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/engine/ErrorPackage.cs [hunk 1]
```diff
@@ -991,13 +991,16 @@ private ErrorRecord()
             object targetObject)
         {
             if (exception == null)
+            {
                 throw PSTraceSource.NewArgumentNullException("exception");
+            }
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/ErrorPackage.cs]: Please use `nameof(exception)`

File: src/System.Management.Automation/engine/ErrorPackage.cs [hunk 2]
```diff
@@ -1071,13 +1074,7 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
         /// Is this instance serialized.
         /// </summary>
         /// <value></value>
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/ErrorPackage.cs]: Is the tag mandatory? Can we remove it?

Comment (vexx32) [reply src/System.Management.Automation/engine/ErrorPackage.cs]: I think it's just automatically added by VS when adding comment XML. I'll remove it. :)

**Comment by vexx32** [comment]
@iSazonov there were a few instances where we could use `nameof()` and a few of the empty `<value></value>` XML tags, so I fixed them all. 🙂

**Review by iSazonov — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**